### ISSUE TITLE
doesn't expand on enter on a newly opened go file

### ISSUE THREAD

**Issue by mortezadadgar** [hrsh7th/nvim-cmp#138]
**test case:**
$ go mod init test
$ touch main.go
$ nvim main.go
try to select `package main` from the menu
like 
![image](https://user-images.githubusercontent.com/27911493/131837855-452514f7-7f5d-4a35-b78e-51fd6428d1cc.png)
it goes to the next line instead of selecting the item

**nvim-cmp config:**
```
local cmp = require "cmp"
local lspkind = require "lspkind"
local keymap = require "cmp.utils.keymap"
local map = cmp.mapping

cmp.setup {
	snippet = {
		expand = function(args)
			vim.fn["vsnip#anonymous"](args.body)
		end,
	},

	mapping = {
		["<C-k>"] = map.select_prev_item(),
		["<C-j>"] = map.select_next_item(),
		['<C-e>'] = cmp.mapping.close(),

		["<Tab>"] = map(function(fallback)
			if vim.fn.pumvisible() == 1 then
				vim.fn.feedkeys(keymap.t "<C-n>", "n")
			elseif vim.fn["vsnip#available"](1) == 1 then
				vim.fn.feedkeys(keymap.t "<Plug>(vsnip-expand-or-jump)", "")
			else
				fallback()
			end
		end, {
			"i",
			"s",
		}),

		["<S-Tab>"] = map(function(fallback)
			if vim.fn.pumvisible() == 1 then
				vim.fn.feedkeys(keymap.t "<C-p>", "n")
			elseif vim.fn["vsnip#available"](-1) == 1 then
				vim.fn.feedkeys(keymap.t "<Plug>(vsnip-jump-prev)", "")
			else
				fallback()
			end
		end, {
			"i",
			"s",
		}),
	},

	sources = {
		{ name = "nvim_lsp" },
		{ name = "buffer" },
		{ name = "vsnip" },
		{ name = "path" },
		{ name = "emoji" },
	},

	-- preselect = cmp.PreselectMode.None,

	completion = {
		completeopt = 'menu,menuone,noinsert',
	},

	formatting = {
		format = function(_, vim_item)
			vim_item.kind = lspkind.presets.default[vim_item.kind] .. ' ' .. vim_item.kind
			return vim_item
		end,
	},
}
``` 

**autopair config:**
```
local npairs = require "nvim-autopairs"
local npairs_cmp = require "nvim-autopairs.completion.cmp"

npairs.setup {
	check_ts = true,
}

npairs_cmp.setup {
	map_cr = true,
	map_complete = true,
}
```

**Comment by hrsh7th** [comment]
@mortezadadgar You should add the `<CR>` mapping.

```lua
cmp.setup {
  mapping = {
    ['<CR>'] = cmp.mapping.confirm()
  }
}
```

**Comment by mortezadadgar** [comment]
@hrsh7th autopair doesn't cover that? it's mentioned in readme

**Comment by hrsh7th** [comment]
I don't know (it is not nvim-cmp ).
Do you get it working?

**Comment by mortezadadgar** [comment]
@hrsh7th yes but I will loss autopair <CR> mapping on opening new pairs, @windwp do you have any idea

**Comment by hrsh7th** [comment]
I hope nvim-autopairs does the mapping to <CR>. (nvim-cmp handles it automatically.)

**Comment by windwp** [comment]
@mortezadadgar  please check reamde  you call a cmp setup first than you call my snippet setup  for cmp
@hrsh7th  

do you want to add an option to add a `(` pair on function method item.
https://github.com/windwp/nvim-autopairs/blob/master/lua/nvim-autopairs/completion/cmp.lua#L28

if you add that i will remove that snippet and only use this code.

```lua
 vim.api.nvim_set_keymap(
            'i',
            '<CR>',
            'v:lua.MPairs.autopairs_cr()',
            { expr = true, noremap = true }
        )
```

so nvim-cmp will handle all another thing.

**Comment by hrsh7th** [comment]
nvim-cmp handles mappings to `<CR>` even if they are already defined.
So the pairs-wise plugin just needs to map to `<CR>` without worrying about nvim-cmp.

(google translated)

**Comment by mortezadadgar** [comment]
@windwp thanks it's working now!

**Comment by windwp** [comment]
@hrsh7th  yes I know it.

but my snippet provides another feature to map_complete item on function item
https://github.com/hrsh7th/nvim-cmp/issues/14#issuecomment-897493673

if you can add this on nvim_cmp so I will remove that on my plugin and i will remove that map_cr too

**Comment by hrsh7th** [comment]
@windwp I thought you are using `event.on_confirm_done` for it. Don't you use it?

**Comment by windwp** [comment]
Yes, it is working now. but I want to ask do you want to add that feature inside nvim-cmp?

then I will remove my code to do that and of course, I will remove that mapping `<cr>` on function too.

**Comment by hrsh7th** [comment]
No. I don't implemente it in nvim-cmp.

I think some features should realize with...

- Confirm completion item
    - It should implemente by `cmp.setup { mapping = { ['<CR>'] = ... } }`
- Fix bracket indentation on `<CR>`
    - It should realize by `nvim-autopairs`.
    - I think `nvim-autopairs` should map to `<CR>` without worrying about nvim-cmp.
        - Because nvim-cmp care about the mapping that already mapped to the buffer.
- Add bracket automatically when confirmed item is `function` or `method` kind
    - It should be realized by `nvim-autopairs`
    - It will be use `cmp.setup { event = { on_confirm_done = function... } }`

**Comment by windwp** [comment]
https://github.com/windwp/nvim-autopairs/blob/master/lua/nvim-autopairs/completion/cmp.lua

please take a look of that link i didn't do anything special on that i just follow what you are talking above
